src/Chocolatey.PowerShell/CONTRIBUTING.md

```diff
@@ -0,0 +1,161 @@
+- Commands that do pipeline processing (declaring a parameter with `[Parameter(ValueFromPipeline = true)]`) will need to handle that input in `Process()`
+- Commands not using pipeline input often handle the bulk of their processing in `End()`
+- If there is any setup that may need to be handled in the command prior to processing any pipeline input, that can go in `Begin()`.
+
+### Logging
+
+Cmdlets inheriting from `ChocolateyCmdlet` will log their parameter values to debug logs when called by default. If any parameters may contain sensitive information, override the `Logging` property and set it to `false` to disable this behaviour.
+
+### Output
+
+By default, `ChocolateyCmdlet`'s `WriteObject(obj)` method will enumerate collections when outputting them, similar to how PowerShell's `Write-Output` works by default.
+If you need to disable this, use the `WriteObject(obj, enumerateCollection: false)` overload.
+
+### Helpers
+
+`ChocolateyCmdlet` provides some helper methods for common operations that might be needed for many cmdlets.
+Some of these (and many more) are also available on the `PSHelper` class.
+
+## Place core logic in helper classes
+
+For more general-purpose PowerShell helpers, add methods to the `PSHelper` so that these can remain in a centralised place.
+
+For other more task-specific helpers:
+- If there is a relevant helper class already present in `src/Chocolatey.PowerShell/Helpers`, add any needed methods to it and have the cmdlet call that method.
+- If there is not already a relevant helper class, add a new one into this folder (and the `Chocolatey.PowerShell.Helpers` namespace).
+
+Unlike in PowerShell functions, C# cmdlets cannot directly call into each other as easily (there is no supported way to instantiate one cmdlet from another in the C# PowerShell API without starting a new subshell and pipeline, which is excessively expensive).
+As such, we need to ensure we leave methods that may need to be shared _not_ on the classes inheriting from `Cmdlet`/`PSCmdlet`/`ChocolateyCmdlet`.
+To work around this, the majority of the core logic of a cmdlet should be placed in a helper class, so it can be easily called from other cmdlets (or helper classes) that may need to reuse the logic.
+
+> :info: **Example**
+> 
+> Take the commands `Install-ChocolateyPackage` and `Install-ChocolateyInstallPackage` for example.
+> `Install-ChocolateyPackage` needs to call `Install-ChocolateyInstallPackage` after downloading its installation files.
+> To facilitate this, we can put the core logic of `Install-ChocolateyInstallPackage` into a helper class, and the actual `InstallChocolateyInstallPackageCommand` class we would write would only define parameters, then call into the helper class.
+> Then, when we write the `InstallChocolateyPackageCommand`, it can download files and then call into the same helper class to run the other cmdlet's logic seamlessly.
+
+### Helpers should return data or throw exceptions, not write output or errors
```

Comment:
For completeness @gep13 could you review this section as it was added after your approval.